andela/rogue-backend
# #167727626 Verify Email upon Signup

#### What does this PR do?
This PR sends a verification email to a new user upon signup into Barefoot Nomad.

#### Description of Task to be completed?
* Write tests that will fail
* create a user model 
* create a verify user model
* create a helper method to use Sendgrid to sent email to user
* Test the user signup endpoint to ensure that a verification email is sent.

#### How should this be manually tested?
* Clone the repo and `cd` into the project directory
* Run `yarn install` to install the project's dependency
* Run `yarn start` to start the app
* Open Postman on `/auth/signup` 
* Pass in the following request parameters  ```firstName, lastName, email, password, isVerified``` as JSON objects
* S
* Enter a valid email address
* Upon signup, user will receiver an email to verify account.

#### What are the relevant pivotal tracker stories?
[##167727626](https://www.pivotaltracker.com/story/show/167727626)

#### Any background context you want to add?
![Screenshot from 2019-08-22 22-08-39](https://user-images.githubusercontent.com/36570152/63550033-a8edd280-c529-11e9-9538-c0b593b1d353.png)
![WhatsApp Image 2019-08-22 at 22 12 51](https://user-images.githubusercontent.com/36570152/63550186-1437a480-c52a-11e9-8afc-8462d50ef400.jpeg)
